src/ResourceManager/Network/Commands.Network/NetworkWatcher/SetAzureNetworkWatcherConfigFlowLogCommand.cs

```diff
@@ -125,6 +245,50 @@ public override void Execute()
                         parameters.RetentionPolicy.Days = this.RetentionInDays;
                     }
 
+                    if (ParameterSetName.Contains(WithTA))
+                    {
+                        parameters.FlowAnalyticsConfiguration = new MNM.TrafficAnalyticsProperties();
+                        parameters.FlowAnalyticsConfiguration.NetworkWatcherFlowAnalyticsConfiguration = new MNM.TrafficAnalyticsConfigurationProperties();
+
+                        parameters.FlowAnalyticsConfiguration.NetworkWatcherFlowAnalyticsConfiguration.Enabled = this.EnableTrafficAnalytics.IsPresent;
+
+                        if (ParameterSetName.Contains(TAByDetails))
+                        {
+                            if (this.WorkspaceResourceId == null || this.WorkspaceGUId == null || this.WorkspaceLocation == null)
```

Comment:
You don't need to have this check - powershell will not allow the user to not enter mandatory parameters for a parameter set.

Reply:
Removed